de.metas.swat/de.metas.swat.base/src/main/java/org/adempiere/invoice/process/CreateAdjustmentChargeFromInvoice.java

```diff
@@ -57,7 +58,8 @@ protected void prepare()
 			}
 			else if (CreateAdjustmentChargeFromInvoice.PARA_DocSubType.equals(name))
 			{
-				parameterDocSubType = (BigDecimal)parameter;
+				// convert the given value of the parameter to BigDecimal
+				parameterDocSubType = NumberUtils.asBigDecimal(parameter, new BigDecimal(-1));
```

Comment:
And more, in case we are not getting rid of prepare() method, i would avoid things like:

`final Object parameter = para[i].getParameter();`
... and then convert to proper type.
Instead, we shall use one of getParameterAsString(), getParameterAsInt(), getParameterAsBoolean(), getParameterAsTimestamp()...